spec/models/comment_spec.rb

```diff
@@ -14,44 +14,56 @@
   end
 
   context "creating a new Comment" do
-    it "#sanitize_body" do
+    it "sanitize the body" do
       body = "hi @#{author.username}. Trying to break comment with <script>alert('bad script')</script>"
       comment = FactoryGirl.create(:comment, body: body)
       expected = "hi @#{author.username}. Trying to break comment with &lt;script&gt;alert(&#39;bad script&#39;)&lt;/script&gt;"
       expect(comment.body).to eq expected
     end
 
-    it "#set_mentions" do
+    it "set the mentions with indices to entities attribute" do
       body = "hi @#{author.username}, @#{author2.username}, and @nonexistent_user"
       comment = FactoryGirl.create(:comment, body: body)
-      expected = {:indices=>[3, 3+('@'+author.username).length]}
+      first_username_length = "@#{author.username}".length
+
+      expected = { "screen_name" => author.username, "indices" => [3, 3 + first_username_length] }
       expect(comment.entities['user_mentions'][0]).to eq expected
-      expect(comment.entities['user_mentions'].length).to eq 2
+      expect(comment.entities['user_mentions'].length).to eq 3
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.